arviz/plots/bf_plot.py

```diff
@@ -0,0 +1,48 @@
+# Plotting and reporting Bayes Factor given idata, var name, prior distribution and reference value
+def plot_bf(trace, var_name, prior, family = 'normal',  ref_val=0, xlim=None, ax=None):
+    # grab trace, a variable name to compute difference and prior.
+    # ref_val is the parameter we want to compare
+    from scipy import stats
+    import matplotlib.pyplot as plt
+    import numpy as np
+    # test some elemtns
+    # varName should be string
+    if not isinstance(var_name, str):
+        print('varName is not a string')
+    # BFs based on density estimation (using kernel smoothing instead of spline)
+    # stack trace posterior
+    tr = trace.posterior.stack(draws=("chain", "draw"))
+    post = tr[var_name]
+    if post.ndim > 1:
+        print("Posterior distribution has {post.ndim} dimensions")
+    if family=='normal':
+        # generate vector
+        if xlim is None:
+            x = np.linspace(np.min(prior), np.max(prior),prior.shape[0])
+        else:
+            x = np.linspace(xlim[0], xlim[1],prior.shape[0])
+        #x = np.linspace(np.min(post), np.max(post),prior.shape[0])
+        my_pdf = stats.gaussian_kde(post)
+        prior_pdf = stats.gaussian_kde(prior)
+    elif family!='normal':
+        # for now and error of notImplemented
+        print("The method for {family} distribution is not implemented yet")
+    if ax is None:
+        fig, ax = plt.subplots()
+    ax.plot(
+        x, my_pdf(x), "--", lw=2.5, alpha=0.6, label="Posterior"
+    )  # distribution function
+    ax.plot(x, prior_pdf(x), "r-", lw=2.5, alpha=0.6, label="Prior")
+    posterior = my_pdf(ref_val) # this gives the pdf at ref_val
```

Comment:
What if `ref_val` is outside the `[min(post), max(post)]` range?

Side note, it looks like https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.gaussian_kde.html has a logpdf method. Having no idea about the expected ranges of values for this variable, could it make sense to compute them in the log space, substract them and exponentiate later for plotting the dot and reporting the values?

Reply:
An interesting question that doesn't seem to be unlikely usually, but I've added an error message if this case happens.

I'm not sure about using logpdf; what is the advantage?